components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/SubscriptionsApiServiceImpl.java

```diff
@@ -270,10 +271,14 @@ public Response subscriptionsSubscriptionIdPut(SubscriptionDTO body, String subs
             String currentThrottlingPolicy = body.getThrottlingPolicy();
             String requestedThrottlingPolicy = body.getRequestedThrottlingPolicy();
 
+            SubscribedAPI subscribedAPI = ApiMgtDAO.getInstance()
+                    .getSubscriptionByUUID(subscriptionId);
             //Check whether the subscription status is not empty and also not blocked
-            if (body.getStatus() != null) {
+            if (body.getStatus() != null && subscribedAPI != null) {
```

Comment:
do we need to do body.getStatus() since there was a condition to check equals

Reply:
if the body.getStatus() is null we are sending error message saying payload should contain subscription status kind of message. if body.getStatus() is not null then we are checking if it's in blocked or on hold status